src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java

```diff
@@ -1331,12 +1331,33 @@ public EntryResult tEncounterActivity2Encounter(
 		// indication -> diagnosis.condition
 		for (Indication cdaIndication : cdaEncounterActivity.getIndications()) {
 			if (!cdaIndication.isSetNullFlavor()) {
-				Condition fhirIndication = tIndication2ConditionEncounter(cdaIndication, bundleInfo);
-				result.addResource(fhirIndication);
-				// TODO: check if this is correct mapping
-				// Reference indicationRef = fhirEncounter.addIndication();
-				// indicationRef.setReference(fhirIndication.getId());
-				fhirEncounter.addDiagnosis().setCondition(getReference(fhirIndication));
+				IEntryResult condResult = tIndication2ConditionEncounter(cdaIndication, bundleInfo);
+
+				result.updateFrom(condResult);
+
+				Condition fhirCondition = null;
+
+				if (condResult.hasResult()) {
+					Bundle bundle = condResult.getBundle();
+					if (bundle != null) {
+						fhirCondition = FHIRUtil.findFirstResource(bundle, Condition.class);
+					}
+
+				} else {
+					Bundle fullBundle = condResult.getFullBundle();
+
+					if (fullBundle != null) {
+						fhirCondition = FHIRUtil.findFirstResource(fullBundle, Condition.class);
+					}
+				}
+
```

Comment:
Can you bake this logic in IEntryResult; maybe a new method that accepts "Condition.class". We are repeating this logic multiple times in multiple places.